src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -456,9 +453,10 @@ impl LogicalOptimizer {
             plan.optimize_by_rules(&GENERAL_UNNESTING_TRANS_APPLY_WITHOUT_SHARE)
         };
         plan = plan.optimize_by_rules_until_fix_point(&GENERAL_UNNESTING_PUSH_DOWN_APPLY);
-        if has_logical_apply(plan.clone()) {
-            return Err(ErrorCode::InternalError("Subquery can not be unnested.".into()).into());
-        }
+
+        // Check if all `Apply`s are eliminated and the subquery is unnested.
+        plan.check_apply_elimination()?;
```

Comment:
After you removed the `to_stream` from logical join, we need to add a visitor (e.g. MaxOneRowFinder) to check whether `MaxOneRow` operators exist for **Streaming Query**. 

The example

```sql
create table t (a int);
explain create materialized view v as  select (select a from t) as c;
```

Reply:
Otherwise, we will meet the error `Not supported: streaming nested-loop join`.

Reply:
You're right. So which approach do you think is better, call `child.to_stream()` first or check `HasLogicalMaxOneRow` before calling `to_stream`? In the latter approach, it seems `LogicalMaxOneRow.to_stream()` can be filled with `unreachable`.

Reply:
This approach looks better IMO:  check HasLogicalMaxOneRow before calling to_stream.